core_lib/data_layers/data/db/sqlalchemy/mixins/soft_delete_mixin.py

```diff
@@ -6,15 +6,7 @@
 class SoftDeleteMixin(object):
     created_at = Column(DateTime, default=datetime.utcnow)
     created_at._creation_order = 9998
-    updated_at = Column(DateTime, default=datetime.utcnow)
+    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

```

Comment:
Thank you this looks much more cleaner. 

please can you share a link to read where `__declare_last__` will not happen all the time.

Reply:
https://docs.sqlalchemy.org/en/14/orm/events.html#sqlalchemy.orm.MapperEvents.before_update

Reply:
thank you and where dose it say the `onupdate` is guaranteed to call?

Reply:
https://docs.sqlalchemy.org/en/14/core/metadata.html?highlight=onupdate#sqlalchemy.schema.Column.params.onupdate